EstateManagement.ContractAggregate.Tests/ContractAggregateTests.cs

```diff
@@ -437,8 +489,85 @@ public void ContractAggregate_AddTransactionFee_VariableValueProduct_InvalidCalc
+        }
+
+        [Theory]
+        [InlineData(CalculationType.Fixed, FeeType.Merchant)]
+        [InlineData(CalculationType.Percentage, FeeType.Merchant)]
+        [InlineData(CalculationType.Fixed, FeeType.ServiceProvider)]
+        [InlineData(CalculationType.Percentage, FeeType.ServiceProvider)]
+        public void ContractAggregate_DisableTransactionFee_TransactionFeeIsDisabled(CalculationType calculationType, FeeType feeType)
+        {
+            ContractAggregate aggregate = ContractAggregate.Create(TestData.ContractId);
+            aggregate.Create(TestData.EstateId, TestData.OperatorId, TestData.ContractDescription);
+            aggregate.AddVariableValueProduct(TestData.ProductId, TestData.ProductName, TestData.ProductDisplayText);
+
+            List<Product> products = aggregate.GetProducts();
+            Product product = products.Single();
+
+            aggregate.AddTransactionFee(product, TestData.TransactionFeeId, TestData.TransactionFeeDescription, calculationType, feeType, TestData.TransactionFeeValue);
+
+            List<Product> productsAfterFeeAdded = aggregate.GetProducts();
+            Product productWithFees = productsAfterFeeAdded.Single();
+            productWithFees.TransactionFees.ShouldHaveSingleItem();
+            TransactionFee fee = productWithFees.TransactionFees.Single();
+            fee.IsEnabled.ShouldBeTrue();
+
+            aggregate.DisableTransactionFee(TestData.ProductId, TestData.TransactionFeeId);
+
+            productsAfterFeeAdded = aggregate.GetProducts();
+            productWithFees = productsAfterFeeAdded.Single();
+            productWithFees.TransactionFees.ShouldHaveSingleItem();
+            fee = productWithFees.TransactionFees.Single();
+            fee.IsEnabled.ShouldBeFalse();
+        }
+
+        [Fact]
+        public void ContractAggregate_DisableTransactionFee_ProductNotFound_ErrorThrown()
+        {
+            ContractAggregate aggregate = ContractAggregate.Create(TestData.ContractId);
+            aggregate.Create(TestData.EstateId, TestData.OperatorId, TestData.ContractDescription);
+
+            Should.Throw<InvalidOperationException>(() =>
+                                                    {
+                                                        aggregate.DisableTransactionFee(TestData.ProductId, TestData.TransactionFeeId);
+                                                    });
+        }
+
+        [Fact]
+        public void ContractAggregate_DisableTransactionFee_TransactionFeeNotFound_ErrorThrown()
+        {
+            ContractAggregate aggregate = ContractAggregate.Create(TestData.ContractId);
+            aggregate.Create(TestData.EstateId, TestData.OperatorId, TestData.ContractDescription);
+            aggregate.AddVariableValueProduct(TestData.ProductId, TestData.ProductName, TestData.ProductDisplayText);
+
+            List<Product> products = aggregate.GetProducts();
+            Product product = products.Single();
```

Comment:
Codacy found an issue: [Remove the unused local variable 'product'.](https://app.codacy.com/manual/stuart_ferguson1/EstateManagement/pullRequest?prid=5995655)